core/src/main/java/com/netflix/msl/io/DefaultMslEncoderFactory.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2017 Netflix, Inc.  All rights reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.msl.io;
+
+import java.io.InputStream;
+import java.util.Set;
+
+/**
+ * <p>Default {@link MslEncoderFactory} implementation that supports the
+ * following encoder formats:
+ * <ul>
+ * <li>JSON: backed by {@code org.json}.</li>
+ * </ul>
+ * </p>
+ * 
+ * @author Wesley Miaw <wmiaw@netflix.com>
+ */
+public class DefaultMslEncoderFactory extends MslEncoderFactory {
```

Comment:
It also creates MslObjects and MslArrays and MslTokenizers. But yes, it does perform operations too. Factory is a misnomer throughout the entire MSL code base right now (as a result of legacy internal discussions with multiple teams back when MSL was being first code reviewed).